src/traces/splom/base_plot.js

```diff
@@ -22,7 +22,7 @@ var SPLOM = 'splom';
 function plot(gd) {
     var fullLayout = gd._fullLayout;
     var _module = Registry.getModule(SPLOM);
-    var splomCalcData = getModuleCalcData(gd.calcdata, _module);
+    var splomCalcData = getModuleCalcData(gd.calcdata, _module)[0];
```

Comment:
@etpinard I changed the API for `getModuleCalcData` in https://github.com/plotly/plotly.js/pull/2561/commits/0b7541e34d63ce74cddb5d243a652b2df0430b70 - so that cartesian could avoid double-drawing the same trace when two trace modules share a plotting step (`box` and `candlestick`). In principle we could extend this pattern (a `plot` step removes the `calcdata` items it plotted and returns the items that are left) which could have a small 🐎 benefit when you have many traces of several types, you don't have to search the whole array each time, only the items that haven't already been handled.

Reply:
:+1: for https://github.com/plotly/plotly.js/commit/0b7541e34d63ce74cddb5d243a652b2df0430b70 

> In principle we could extend this pattern (a plot step removes the calcdata items it plotted and returns the items that are left) which could have a small racehorse benefit when you have many traces of several types, you don't have to search the whole array each time, only the items that haven't already been handled.

Good idea here. We should keep that in mind when we'll :hocho: `calcdata` completely.